src/Microsoft.ML.Data/DataLoadSave/Database/DatabaseSource.cs

```diff
@@ -25,6 +25,21 @@ public DatabaseSource(DbProviderFactory providerFactory, string connectionString
             CommandText = commandText;
         }
 
+        /// <summary>Creates a new instance of the <see cref="DatabaseSource" /> class.</summary>
+        /// <param name="providerFactory">The factory used to create the <see cref="DbConnection"/>..</param>
+        /// <param name="connectionString">The string used to open the connection.</param>
+        /// <param name="commandText">The text command to run against the data source.</param>
+        /// <param name="commandTimeOutInSeconds">The timeout(in seconds) for database command.</param>
+        public DatabaseSource(DbProviderFactory providerFactory, string connectionString, string commandText, int commandTimeOutInSeconds) :
+            this(providerFactory, connectionString, commandText)
+        {
+            Contracts.CheckUserArg(commandTimeOutInSeconds >= 0, nameof(commandTimeOutInSeconds));
+            CommandTimeOutInSeconds = commandTimeOutInSeconds;
+        }
+
+        /// <summary>Gets the timeout for database command.</summary>
+        public int? CommandTimeOutInSeconds { get; }
+
```

Comment:
nit: Within computing (and to some extent in general English) `Timeout` is considered a single word with its own meaning and so you strictly don't need to uppercase the O as in `TimeOut`